### PR TITLE
[#33][Integrate] As a user, I can submit my survey response

### PR DESCRIPTION
**Pull request by hoangnguyen92dn**
Close https://github.com/hoangnguyen92dn/survey-flutter-ic/issues/33

## What happened 👀

Implement submit survey to API endpoint

## Insight 📝

- Cache the selected answer to ViewModel then submit the API request at the last step
- Update Unit and Ui Test

## Proof Of Work 📹

https://user-images.githubusercontent.com/6950766/234068996-87d70e4d-1471-418a-b7a7-02b3ff6b8e6a.mp4

- UnitTest passed ✅ 
image

- IntegrationTest passed ✅ 
image

### CODE REVIEW

File: lib/widget/answer_textarea.dart [hunk 1]
```diff
@@ -1,21 +1,45 @@
...
+  });
 
   @override
-  State<AnswerTextarea> createState() => _AnswerTextareaState();
+  ConsumerState<AnswerTextarea> createState() => _AnswerTextareaState();
 }
 
-class _AnswerTextareaState extends State<AnswerTextarea> {
+class _AnswerTextareaState extends ConsumerState<AnswerTextarea> {
+  final _controller = TextEditingController();
```
Comment (manh-t) [inline lib/widget/answer_textarea.dart:26]: missing `dispose` for this controller 🙋

File: lib/widget/answer_dropdown.dart [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
   @override
   Widget build(BuildContext context) {
+    ref.listen(nextQuestionStream, (previous, next) {
+      ref
+          .watch(surveyQuestionsViewModelProvider.notifier)
+          .cacheAnswers(SubmitSurveyQuestionsRequest(
+            questionId: widget.questionId,
+            answers: [
+              SubmitSurveyAnswersRequest(
+                  answerId: widget.answers[_selectedAnswerIndex + 1].id)
```
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply lib/widget/answer_dropdown.dart]: @manh-t It's wrong here 😮‍💨 The index always returns 0, I updated the way to get the selected answer in here [e80dbd1](https://github.com/hoangnguyen92dn/survey-flutter-ic/pull/73/commits/e80dbd15fd429c558ea4321005544d1583844275) 🚀 

```
      onSelect: (Picker picker, int index, List value) {
        final selectedAnswer =
            picker.getSelectedValues()[0] as SurveyAnswerUiModel;
        _selectedAnswerId = selectedAnswer.id;
      },
```

File: lib/ui/questions/survey_questions_screen.dart [hunk 3]
```diff
@@ -154,26 +155,37 @@ class _SurveyQuestionsScreenScreenState
...
+                FlatButtonText(
+                  key: SurveyQuestionsWidgetId.submitButton,
+                  text:
+                      context.localization.survey_details_submit_survey_button,
+                  isEnabled: true,
+                  onPressed: () => {
+                    ref
+                        .read(surveyQuestionsViewModelProvider.notifier)
+                        .submit(widget.surveyId)
+                  },
```
Comment (manh-t) [inline lib/ui/questions/survey_questions_screen.dart:183]:
```suggestion
                  onPressed: () {
                    ref
                        .read(surveyQuestionsViewModelProvider.notifier)
                        .submit(widget.surveyId);
                  },
```
please use the callback function consistently, we should use:
- `() { ... }`
- `() => ...`
NOT:
- `() => { ... }`
😓

**Review by manh-t — COMMENTED** [review]
rest LGTM 🚀

**Review by manh-t — APPROVED** [review]

**Pull request merged**